### PR TITLE
refactor: Standardize TensorFlowNET.Keras/Losses/*

### PR DESCRIPTION
**Pull request by DevNullx64**
abstract Loos.Apply instead of virtualy throw exception
abstract class LossFunctionWrapper
Smooth implementation of sub class
Move ILossFunc to base abstract class Loss

### CODE REVIEW

**Comment by Oceania2018** [comment]
Why do we remove `ILossFunc`? It's designed to decouple the `Tensorflow.Binding` with `Tensorflow.Keras` projects.
`ILossFunc` is the standard interface between the two projects.
I think what we need to do is changing all the `loss` function to implement the interface, then in `Tensorflow.Binding` can use `Keras` loss functions.

**Comment by DevNullx64** [comment]
I've put the ILossFunc implementation on base Loss abstract class. Everithing that inherit from Loss inherit ILossFunc.
